aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -848,6 +864,117 @@ impl AptosVM {
         Ok(())
     }
 
+    fn validate_and_execute_batched_calls(
+        &self,
+        resolver: &impl AptosMoveResolver,
+        session: &mut SessionExt,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext,
+        senders: Vec<AccountAddress>,
+        batched_calls: &[BatchedFunctionCall],
+        txn_data: &TransactionMetadata,
+    ) -> Result<(), VMStatus> {
+        let mut batched_return_values = BatchedExecutionResult {
+            return_values: vec![],
+        };
+        for batched_call in batched_calls {
+            self.validate_and_execute_batched_call(
+                resolver,
+                session,
+                gas_meter,
+                traversal_context,
+                senders.clone(),
+                batched_call,
+                &mut batched_return_values,
+                txn_data,
+            )?;
+        }
+        for return_values in batched_return_values.return_values {
+            for value in return_values {
+                if !value.ability.has_drop() && !value.consumed {
```

Comment:
Why? I can in theory pass the returned value into the next batched call, without dropping it. This would be possible with scripts?

Reply:
This is for the ability safety: towards the end of txn execution, all values either need to be consumed by passing it to the next batched call, or is droppable.